aten/src/ATen/native/native_functions.yaml

```diff
@@ -8807,6 +8807,26 @@
   dispatch:
     CPU, CUDA: linalg_vector_norm_out
 
+- func: linalg_matrix_norm(Tensor self, Scalar? ord=None, int[]? dim=None, bool keepdim=False, *, ScalarType? dtype=None) -> Tensor
+  python_module: linalg
+  dispatch:
+    CPU, CUDA: linalg_matrix_norm
+
+- func: linalg_matrix_norm.out(Tensor self, Scalar? ord=None, int[]? dim=None, bool keepdim=False, *, ScalarType? dtype=None, Tensor(a!) out) -> Tensor(a!)
+  python_module: linalg
+  dispatch:
+    CPU, CUDA: linalg_matrix_norm_out
```

Comment:
Please consider making this structured! https://github.com/pytorch/rfcs/blob/rfc-0005/RFC-0005-structured-kernel-definitions.md

Reply:
@ezyang I will do this in the next PR which will move the implementation to `linalg.matrix_norm` and deprecate `linalg.norm`. Currently, the shape checking logic is in `linalg.norm`.